### PR TITLE
#160617670 add functionality to reset password via email

### PR DESCRIPTION
**Pull request by kimbugp**
### What does this PR do?
- Implement the feature to update user password using an email
#### Description of Task to be completed?
- When a user forgets their password, they enter their email address 
- An email is sent to them with a reset link
- The reset link provides them with a token and URL.
- Then they are allowed to change password.
#### How should this be manually tested?
- Use the  route ending `api/users/password_reset` to send email data in the format `'user': {
                'email': email
            }`
- Use the link send to the email for details
#### Screenshots
- Send email url

send email

- Link sent to email

email link

- Response from email link

token return

- Token added to header on reset path

header

- Password succefully changed

rest

#### What are the relevant pivotal tracker stories?
- #160617670

### CODE REVIEW

**Review by PatrickCmd — COMMENTED** [review]
@kimbugp Please do include screenshoots of how you test this endpoint in postman at the various stages.

File: authors/apps/authentication/serializers.py [hunk 1]
```diff
@@ -143,3 +144,30 @@ def update(self, instance, validated_data):
...
+            raise serializers.ValidationError(
+                'An email is required to reset password.'
+            )
+        user = get_object_or_404(User, email=email)
+        # user.is_active=False
+        
+        # return token here
+        return {
+            'email': user,
+            'token':user.token(0.0208333),
```
Comment (AmosWels) [inline authors/apps/authentication/serializers.py:212]: Does this token override the JWT token payload that is generated in the models?

Comment (kimbugp) [reply authors/apps/authentication/serializers.py:212]: the method to create a token in the models allows you to change the default value of the number of days the token is valid.  So by turning 30 minutes into days, I can change how long the token generated is valid

**Review by AmosWels — APPROVED** [review]
LGTM

**Comment by kimbugp** [comment]
@PatrickCmd ,I have added the required screenshots

**Pull request merged**